client/src/Components/Main/index.ts

```diff
@@ -2,121 +2,49 @@ import Component from '@/Core/Component';
 import './styles';
 import { Props, State } from '@/utils/types';
 import { html } from '@/utils/helper';
-import { histories } from '@/assets/dummy';
-import List from '@/Components/List';
+import MainModel from '@/Model/MainModel';
+import HistoryDayCard from '../HistoryDayCard';
+import { IHistory } from '@/utils/types';
+
+interface IMainState extends State {
+  historyCards: IHistory[];
+}
+
+export default class Main extends Component<IMainState, Props> {
+  model: any;
+
+  setup() {
+    this.model = MainModel;
+    this.model.subscribe('history', 'MAIN', this);
```

Comment:
`unmount` 시점때 `unsubscribe` 호출하면서 기존에 중복검사 로직을 제거했습니당~~

그래서 컴포넌트 이름은 이제 따로 전달해주지 않아도 되어요~

그리고 `key` 값의 경우도 모델 자체에 생성해두고

이를 구독하는 뷰의 입장에서는 별도로 키를 입력하지 않고 바로 모델 키 값을 사용하는 식으로 사용하고 있는데,

이는 저번에 이야기 했던 거 처럼 아예 키 값을 입력하지 않아도 알아서 등록하도록 바꾸어 볼 여지가 있을 거 같습니당~~

```suggestion
    this.model.subscribe('this.model.key', this);
```